File: c/cs50/mario.c
#include <stdio.h>
#include <cs50.h>

int get_input(void);

int main(void)
{
    int n = get_input();
    for (int i = 0; i < n; i++) //row iterator
    {
        for (int j = 0; j <= 2 * n + 1; j++) //column iterator
        {
            if (i + j < n - 1 || j == n || j == n + 1) //overrides with spaces
            {
                printf(" ");
            }
            else if (!(j - i > n + 2)) //print hash everywhere else except once pyramid is completed
            {
                printf("#");
            }
        }
        printf("\n");
    }
}

int get_input(void)
{
    int n;
    do //repeatedly asks for input until a number between 1 and 9 inclusive is given
    {
        n = get_int("Height: ");
    }
    while (n < 1 || n > 8);
    return n;
}

File: python/cs50/mario.py
#!/usr/bin/env python3
from cs50 import get_int

n = 0
while n not in range(1, 9):
    n = get_int("Height: ")

for i in range(n):  # row iterator
    for j in range(2 * n + 2):  # column iterator
        if i + j < n - 1 or j == n or j == n + 1:  # overrides with spaces
            print(end=" ")
        elif not j - i > n + 2:  # print hash everywhere else except once pyramid is completed
            print(end="#")
    print()